### PR TITLE
Move to Bootstrap 5

### PR DESCRIPTION
**Pull request by alvinBM**
## Pullrequest
Move to bootstrap 5

## Issues
- #762

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -38,7 +38,9 @@
         "webklex/composer-info": "^0.0.1",
         "composer" : "^2.0.0",
         "zbateson/mail-mime-parser": "^2.4",
-        "league/commonmark": "^2.4"
+        "league/commonmark": "^2.4",
+        "twbs/bootstrap": "5.0.2",
```
Comment (marclaporte) [inline composer.json:42]: 5.3.2 is most recent release
^ or ~ should be used to automate

Comment (alvinBM) [reply composer.json]: Upgraded. Thanks

**Comment by marclaporte** [comment]
I am thrilled there is progress here!

**Comment by marclaporte** [comment]
yay!

File: modules/calendar/modules.php [hunk 2]
```diff
@@ -196,9 +231,9 @@ protected function output() {
             $events = $cal_events->in_date_range($bounds[0], $bounds[1]);
             $out = new Hm_Cal_Output($this, $events);
             $out = $out->output($data, $date, $view);
-            return '<div class="calendar"><div class="content_title">'.$this->trans('Calendar').'</div>'.
-                '<div class="list_controls"><a href="?page=calendar&amp;action=add" title="'.$this->trans('Add Event').'" class="refresh_list">'.
```
Comment (kroky) [inline modules/calendar/modules.php:200]: Classes like list_controls are used in JS context. Did you check all the classes you removed that they are only used for CSS styling and not anywhere in the JS code?

File: modules/core/site.css [hunk 3]
```diff
@@ -7,8 +12,8 @@ html, body { max-width: 100%; min-height: 100%; background-color: #fff; }
 body { background: linear-gradient(180deg, #faf6f5, #faf6f5, #faf6f5, #faf6f5, #fff); font-size: 1em; height: 100%; color: #333; font-family: Arial; padding: 0px; margin: 0px; min-width: 700px; font-size: 100%; }
 table { width: 100%; border-spacing: 0px; }
 td { vertical-align: top; }
-.adv_controls .combined_sort, .message_list .combined_sort { font-size: 80%; margin-top: -2px; ; float: right; margin-right: 125px; }
-.adv_controls .combined_sort { margin-right: 20px; }
+/* .adv_controls .combined_sort, .message_list .combined_sort { font-size: 80%; margin-top: -2px; ; float: right; margin-right: 125px; } */
```
Comment (kroky) [inline modules/core/site.css:15]: Maybe it is temporary but we want to get rid of the comments in this file.

File: modules/pop3/modules.php [hunk 4]
```diff
@@ -0,0 +1,1129 @@
+<?php
```
Comment (kroky) [inline modules/pop3/modules.php:1]: I think we removed the pop3 module. No need for converting it to bootstrap.

**Comment by kroky** [comment]
That seems like quite an effort. I added just a couple of comments. Code looks good. I think more people should test each form/table/page in Cypht to ensure it works correctly. Also, after that's merged, we need to test the Tiki integration as there might need to be some tweaks (especially if bootstrap versions are different).

File: modules/pop3/modules.php (hunk 4 above)
Comment (marclaporte) [reply modules/pop3/modules.php:1]: Indeed: https://github.com/cypht-org/cypht/pull/614

Any POP leftovers should be removed.

**Comment by marclaporte** [comment]
At a meeting today, @alvinBM indicated that future steps (in distinct PRs) include adding some themes from Bootswatch and testing the integration in Tiki.

**Comment by josaphatim** [comment]
@alvinBM I just found some points that need to be fixed.

- Login and success buttons color should be teal to make it uniform with link I guess.
- Remove css styling of .nux_password
- Add bootstrap to .credentials, .search_terms
- Test, delete, forget, hide buttons are not working anymore on /?page=servers page
- Class Hm_Image_Sources should be deleted as icons are replaced by twbs/bootstrap-icons
- Adding servers (IMAP, JMAP, SMTP) is not working
- Delete button on /?page=contacts page is not working
- Many tests are failing
- Review padding on /?page=highlights page when there is no rule yet
- Add bootstrap to Github, 2fa and Wordpress Settings (Activate those modules) then got to /?page=settings page
- Add bootstrap to 2fa authentication page

**Comment by josaphatim** [comment]
@alvinBM I think also:
- Inputs and buttons on modal Add script and add filter must also have bootstrap classes.
- Page /?page=block_list has no bootstrap yet
- On message list page:
       - Snooze button dropdown must be replaced by bootstrap dropdown
       - copy and move button are not working and must all toggle a bootstrap dropdown, our custom dropdowns must be replaced everywhere.
- On message view page: 
      - Block sender button dropdown must also be replaced and its inputs must use bootstrap classes
      - In from header: there is a button that toggle a dropdown must be replaced too.
      - Next and previous buttons icons must be replaced too
- Class .unread_count must also have a .badge class

**Comment by alvinBM** [comment]
> @alvinBM I just found some points that need to be fixed.
> 
> * Login and success buttons color should be teal to make it uniform with link I guess.
> * Remove css styling of .nux_password
> * Add bootstrap to .credentials, .search_terms
> * Test, delete, forget, hide buttons are not working anymore on /?page=servers page
> * Class Hm_Image_Sources should be deleted as icons are replaced by twbs/bootstrap-icons
> * Adding servers (IMAP, JMAP, SMTP) is not working
> * Delete button on /?page=contacts page is not working
> * Many tests are failing
> * Review padding on /?page=highlights page when there is no rule yet
> * Add bootstrap to Github, 2fa and Wordpress Settings (Activate those modules) then got to /?page=settings page
> * Add bootstrap to 2fa authentication page

Thank you for your feedback @josaphatim,  Work in progress on all these proposals

File: modules/calendar/modules.php (hunk 2 above)
Comment (alvinBM) [reply modules/calendar/modules.php:200]: Thanks @kroky , 
I took care not to delete the classes that were referenced by the js scripts. For this case, there was no code breakage

File: modules/pop3/modules.php (hunk 4 above)
Comment (alvinBM) [reply modules/pop3/modules.php:1]: All right, Thanks

File: language/fa.php [hunk 5]
```diff
@@ -249,7 +249,7 @@
...
+    "Settings are not saved permanently on the server unless you explicitly allow it. If you don\'t save your settings, any changes made since you last logged in will be deleted when your session expires or you logout. You must re-enter your password for security purposes to save your settings permanently."=>"تنظیمات برای همیشه در سرور ذخیره نمی شوند ، مگر اینکه صریحاً اجازه آن را بدهید. اگر تنظیمات خود را ذخیره نکنید ، با اتمام جلسه یا خارج شدن از سیستم ، هر تغییری که از آخرین ورود به سیستم انجام داده اید ، حذف خواهد شد. برای ذخیره تنظیمات خود باید دوباره رمز ورود خود را برای اهداف امنیتی وارد کنید.",
```
Comment (marclaporte) [inline language/fa.php:252]: Is this intentional?

File: modules/2fa/site.css [hunk 6]
```diff
@@ -1,3 +1,4 @@
+<<<<<<< HEAD
```
Comment (marclaporte) [inline modules/2fa/site.css:1]: gremlin

File: modules/2fa/site.css [hunk 7]
```diff
@@ -216,4 +217,7 @@
 }
 /**
 	**		Shake Annimation Ends Here.
-	**/
\ No newline at end of file
+	**/
+=======
+.tfa_setting { display: none; }
+>>>>>>> ba2058f0 (2fa Module)
```
Comment (marclaporte) [inline modules/2fa/site.css:223]: gremlin

File: language/fa.php (hunk 5 above)
Comment (alvinBM) [reply language/fa.php:252]: Yes

**Comment by alvinBM** [comment]
> * Many tests are failing

I fixed most of the tests, and I noticed that others were not even linked to my PL, and even in master I have quite a few tests that fail.

**Comment by alvinBM** [comment]
Hello @marclaporte, @kroky, @josaphatim , I think that this PL is ready at this level, and as I said previously, the work of migrating to bootstrap 5 and improving the UI will be continuous work, so merging this one will allow other person who works on UI/UX to start with a new bootstrap5 base, and that will save me too many conflicts every time I try to be in sync with the master.

**Comment by kroky** [comment]
Alright, merging this and we will resolve further issues with different PRs...

**Comment by marclaporte** [comment]
Bootswatch is coming: https://github.com/cypht-org/cypht/pull/921

**Pull request merged**